### PR TITLE
macOS dual arch (Intel & M1) builds with Carthage

### PR DESCRIPTION
**Pull request by petrmifek**
Recently (~XCode 12.5) Mantle was preventing our project to archive due to a missing/disabled arm architecture support on **macOS** (debugging on Intel works, but dual architecture **archiving for Intel and M1 fails**). Mantle Carthage project for macOS is also using the **VALID_ARCHS** setting for that restriction, which **has been deprecated since XCode 12**.

Here are few quick fixes and updates we had to make:
- update the xcconfigs to latest (as a "side effect" also update the Nimble and Quick submodules to match Cartfile.resolved)
- switch the xcconfigs from "Mac OS" to the new "macOS" variants

These changes then required (likely due a stricter defaults in xcconfigs) a few changes to the code - mostly mismatched nullability issues. They are separated in commits, may be that the last one regarding the class_getProperty should go to a macro somewhere, as it repeats those lines like 5x. If so, I'd be happy to add that if someone could point me a suitable header file where to put those.

Thanks!

### CODE REVIEW

**Comment by robb** [comment]
Hi, thanks for looking into this.

There seems to be a white space issue with the `class_getProperty` changes – Mantle uses tabs.

Also, do you think you could investigate the  tests failing? A quick look makes it seem like Nimble and Quick aren't building.

**Comment by petrmifek** [comment]
Ah, thanks, thought that was just my local issue the tests failing, unfortunately that wasn't the case :)

Please try now. Quick and Nimble needed to be upgraded plus a few things similar to the main code.

File: Mantle/MTLJSONAdapter.m [hunk 1]
```diff
@@ -356,7 +356,7 @@ - (id)modelFromJSONDictionary:(NSDictionary *)JSONDictionary error:(NSError **)e
 				NSDictionary *userInfo = @{
 					NSLocalizedDescriptionKey: [NSString stringWithFormat:@"Caught exception parsing JSON key path \"%@\" for model class: %@", JSONKeyPaths, self.modelClass],
 					NSLocalizedRecoverySuggestionErrorKey: ex.description,
-					NSLocalizedFailureReasonErrorKey: ex.reason,
+                    NSLocalizedFailureReasonErrorKey: ex.reason == nil ? @"" : ex.reason,
```
Comment (robb) [inline Mantle/MTLJSONAdapter.m:359]: Can I bother you with one more whitespace pass? Xcode used to pick up the tabs preference from the project file, not sure if that broke…

```suggestion
					NSLocalizedFailureReasonErrorKey: ex.reason == nil ? @"" : ex.reason,
```

File: Mantle/NSError+MTLModelException.m [hunk 2]
```diff
@@ -26,7 +26,7 @@ + (instancetype)mtl_modelErrorWithException:(NSException *)exception {
 
 	NSDictionary *userInfo = @{
 		NSLocalizedDescriptionKey: exception.description,
-		NSLocalizedFailureReasonErrorKey: exception.reason,
+        NSLocalizedFailureReasonErrorKey: exception.reason == nil ? @"" : exception.reason,
```
Comment (robb) [inline Mantle/NSError+MTLModelException.m:29]:
```suggestion
		NSLocalizedFailureReasonErrorKey: exception.reason == nil ? @"" : exception.reason,
```

Comment (robb) [reply Mantle/NSError+MTLModelException.m:29]: Also, I think you can use the `?:` pattern to get rid of `nil`s here.

```suggestion
		NSLocalizedFailureReasonErrorKey : exception.reason ?: @"",
```

File: MantleTests/MTLTestModel.m [hunk 3]
```diff
@@ -113,8 +113,8 @@ + (NSDictionary *)dictionaryValueFromArchivedExternalRepresentation:(NSDictionar
 	NSParameterAssert(fromVersion == 1);
 
 	return @{
-		@"name": externalRepresentation[@"username"],
-		@"nestedName": externalRepresentation[@"nested"][@"name"],
+		@"name": externalRepresentation[@"username"] == nil ? [NSNull null] : externalRepresentation[@"username"],
+		@"nestedName": externalRepresentation[@"nested"][@"name"] == nil ? [NSNull null] : externalRepresentation[@"nested"][@"name"],
```
Comment (robb) [inline MantleTests/MTLTestModel.m:117]: As above.

```suggestion
		@"name": externalRepresentation[@"username"] ?: [NSNull null],
		@"nestedName": externalRepresentation[@"nested"][@"name"] ?: [NSNull null],
```

**Review by robb — CHANGES_REQUESTED** [review]
Looking good, just a few minor nits.

**Comment by robb** [comment]
Seems like the tests a going to base (2/5 still in progress as of now…)

**Comment by petrmifek** [comment]
Thanks! Here you go... Hopefully this time all the spaces are away. Must've confused it with some copy-pastes or what...

**Review by robb — APPROVED** [review]
Looks good! 💖

**Pull request merged**